File: ZTTextFiled-HybirdDemo/ZTTextFieldText/ViewController.m
//
//  ViewController.m
//  ZTTextFieldText
//
//  Created by zhaitong on 2017/12/14.
//  Copyright © 2017年 zhaitong. All rights reserved.
//

#import "ViewController.h"
#import "ZTTextFieldText-Swift.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet ZTTextField *textField;
@property (weak, nonatomic) IBOutlet ZTTextField *noField;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.textField becomeFirstResponder];

    _textField.validationCloure = ^NSString * _Nonnull(ZTTextField * _Nonnull textField) {
        if (textField.text.length > 0) {
            return @"姓名不能为空";
        }
        return @"";
    };
}
- (IBAction)didPressedDone:(id)sender {

    [self.textField textFieldShowWrongMessageWithSWrongMessage:@"姓名不能为空"];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];

}


@end

File: ZTTextField-Swift/ZTTextField-Swift/ViewController.swift
//
//  ViewController.swift
//  ZTTextField-Swift
//
//  Created by zhaitong on 2017/12/22.
//  Copyright © 2017年 zhaitong. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var textFieldName: ZTTextField!
    @IBOutlet weak var textFieldMobile: ZTTextField!
    @IBOutlet weak var textFieldCertyNo: ZTTextField!
    override func viewDidLoad() {
        super.viewDidLoad()
        self.textFieldName.becomeFirstResponder()
        let closure = { (textField : ZTTextField) -> String in
            
            if let text = textField.text {
                guard text.count > 0 else {
                    return ""
                }
                return "姓名不能为空"
            }
            return ""
        }
        self.textFieldName.validationCloure = closure
        self.textFieldCertyNo.textFieldType = .ZTTextFieldTypeNormal
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

